File: Graph Traversal/1325.py
N,M=map(int, input().split())
G,O=[[] for _ in range(N+1)],[0 for _ in range(N+1)]
for i in range(M):
    A,B=map(int, input().split())
    G[B].append(A)
def dfs(s):
    V,S,o=[0 for _ in range(N+1)],[s],1
    V[s]=1
    while len(S)>0:
        c=S.pop()
        for n in G[c]:
            if not V[n]: 
                S.append(n)
                V[n]=1
                o+=1
    return o
m=0
for i in range(1,N+1):
    O[i]=dfs(i)
    m=max(m,O[i])
for i in range(1,N+1):
    if O[i]==m: print(i,end=' ')

File: Graph Traversal/1325.js
const [[N], ...I] = `${require('fs').readFileSync('./dev/stdin')}`.trim()
  .split`\n`.map(e => e.split` `.map(Number))
const G = [...Array(N + 1)].map(_ => [])
I.forEach(([A, B]) => G[B].push(A))
let [max, O] = [0, []]
for (let i = 1; i <= N; i++) {
  let [V, o] = [Array(N + 1).fill(0), 1]
  const dfs = c => {
    V[c]++
    for (let i = 0; i < G[c].length; i++) {
      const n = G[c][i]
      !V[n] && ++o && dfs(n, V)
    }
  }
  dfs(i, V)
  max == o && O.push(i)
  max < o && (O = [i]) && (max = o)
}
console.log(O.join` `)
